### ISSUE TITLE
MediaLibraryField usage cause error in 5.3+ (migration to TS)

### ISSUE THREAD

**Issue by cyp3rius** [strapi/strapi#22258]
### Required System information

- Node.js version: v20.12.0
- NPM version:  10.5.0
- Strapi version: 5.4.1 (+ Yarn 4.4.1)
- Database: SQLite
- Operating system: OSX
- Is your project Javascript or Typescript: TypeScript

### Describe the bug

Some of the plugins we're developing (`reactions` & `navigation`) use the `MediaLibraryField` exposed by Strapi. Till version 5.3.x the implementation following by migration guidelines:

```ts
const fields = useStrapiApp('ReactionsPluginSettings', (state) => state.fields);
```

does not cause any issues. Since version 5.3+ so when you migrated the Media Library to TS, following error pops and block the UI.

```
TypeError: Cannot read properties of undefined (reading 'allowedTypes')
    at http://localhost:1337/node_modules/.strapi/vite/deps/chunk-MZGQORAC.js?v=7827988d:15044:18
    at renderWithHooks (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:11568:26)
    at updateForwardRef (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:14345:28)
    at beginWork (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:15966:22)
    at beginWork$1 (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:19781:22)
    at performUnitOfWork (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:19226:20)
    at workLoopSync (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:19165:13)
    at renderRootSync (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:19144:15)
    at recoverFromConcurrentError (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:18764:28)
    at performSyncWorkOnRoot (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-YCN7MIJY.js?v=7827988d:18907:28)
```

<img width="1439" alt="Screenshot 2024-11-24 at 17 18 58" src="https://github.com/user-attachments/assets/6e2d1df3-8894-4199-9899-50b6d7ef6cbf">

### Steps to reproduce the behavior

0. Install the latest version of `reactions` plugin in your 5.3+ instance of Strapi
1. Go to Settings -> Reactions
2. Click on `Add new reaction`
3. See error

### Expected behavior

The modal with Media Library Field should appear

**Comment by simotae14** [comment]
Hello @cyp3rius the issue will be solved in the next version of Strapi, sorry for the incovenience

**Comment by simotae14** [comment]
Hello @cyp3rius my fix is part of the v5.4.2 release.
We tried to use your pluign in the Edit view without success, even with 5.0.0 (without media library migration), can I ask you to check if you need more help?
Thanks in advance
